flink-core/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractor.java

```diff
@@ -1900,7 +1900,8 @@ private boolean isValidPojoField(Field f, Class<?> clazz, ArrayList<Type> typeHi
 			ParameterizedType parameterizedType, TypeInformation<IN1> in1Type, TypeInformation<IN2> in2Type) {
 
 		if (!Modifier.isPublic(clazz.getModifiers())) {
-			LOG.info("Class " + clazz.getName() + " is not public, cannot treat it as a POJO type. Will be handled as GenericType");
+			LOG.info("Class " + clazz.getName() + " is not public, cannot treat it as a POJO type. " +
+				"Will be handled as GenericType, and may lose some serialization performance.");
```

Comment:
What if we said something like "... is not public and cannot be processed as a POJO type. Please read the Flink documentation on "Data Types & Serialization" for details on the impact to performance."